h2integrate/postprocess/mapping.py

```diff
@@ -0,0 +1,852 @@
+            longitude data as defined in the .csv or cases.sql file. Defaults to None.
+            The code will attempt to automatically detect this column if no string is provided and
+            will raise an error if unable to resolve to one column.
+        fig (plt.Figure, optional): A plt.Figure object of an existing map on which to add layers.
+            Defaults to None.
+        ax (plt.Axes, optional): A plt.Axes object of an existing map on which to add layers.
+            Defaults to None.
+        base_layer_gdf (
+            gpd.GeoDataFrame | list[gpd.GeoDataFrame] | tuple[gpd.GeoDataFrame, ...],
+            optional
+        ):
+            One or more GeoDataFrame objects previously used to create the base map. These are used
+            to validate CRS consistency and to calculate map bounds when adding new layers.
+            Defaults to None.
+        show_plot (bool, optional): A boolean to control whether the plot is displayed. When adding
+            multiple layers to the same figure, this should be set to False until the final layer
+            is added. Defaults to False.
+        save_plot_fpath (Path | str, optional): A string or Path object specifying where to save the
+            generated plot. If None, the plot is not saved. Defaults to None.
+        map_preferences (dict | GeospatialMapConfig, optional): A dictionary or GeospatialMapConfig
+            object defining formatting and plotting preferences for GeoPandas, matplotlib, and
+            contextily. Defaults to {}.
+        save_sql_file_to_csv (bool, optional): A boolean used to save the results from cases.sql
+            files into a csv file. passed into the convert_sql_to_csv_summary() function.
+            Defaults to False.
+
+    Returns:
+        fig (plt.Figure): The plt.Figure object for the current plot. This can be passed back into
+            the function to add additional layers.
+        ax (plt.Axes): The plt.Axes object for the current plot. This can be passed back into the
+            function to add additional layers.
+        results_gdf (gpd.GeoDataFrame): The gpd.GeoDataFrame object created from the parsed results.
+            This can be passed back into the function to add additional layers.
+
+    Raises:
+        TypeError: If the provided case_results_fpath is of the wront type (not .csv or .sql)
+        ValueError: If only a subset of fig, ax, and base_layer_gdf is provided when adding a layer.
+    """
+    # Check if map_preferences is a dictionary and load as GeospatialMapConfig.from_dict()
+    # Loads default values and checks for any erroneous / extraneous keys
+    if isinstance(map_preferences, dict):
+        map_preferences = GeospatialMapConfig.from_dict(map_preferences, strict=True)
+
+    # Load data
+    case_results_fpath = find_file(case_results_fpath)
+
+    # If case_results_fpath is a .csv file read in with pandas
+    if (".csv") in case_results_fpath.suffix:
```

Comment:
FYI, length-1 tuples get reduced to the single object.
```suggestion
    if ".csv" in case_results_fpath.suffix:
```